src/ex_color/model.py

```diff
@@ -25,11 +46,104 @@ def __init__(self):
             nn.Sigmoid(),  # Keep RGB values in [0,1]
         )
 
-    # TODO: Don't return latents from here: use hooks or callbacks instead
-    def forward(self, x: torch.Tensor) -> tuple[torch.Tensor, torch.Tensor]:
+        # Register forward hook to capture latents
+        self.encoder.register_forward_hook(self._capture_latents_hook)
+
+    def _capture_latents_hook(self, module, input, output):
+        """Hook to capture latent representations from the encoder."""
+        self.current_latents = output
```

Comment:
This is cool, but is it possible to capture the latents _outside_ the model? Like, with a hook class or something?